bokeh/core/property/wrappers.py

```diff
@@ -407,12 +407,15 @@ def _stream(self, doc, source, new_data, rollover=None, setter=None):
         """
         old = self._saved_copy()
 
+        conditions = [isinstance(self[k], np.ndarray) or isinstance(new_data[k], np.ndarray)
+            or isinstance(self[k], pd.core.series.Series)
+            or isinstance(new_data[k], pd.core.series.Series)]
```

Comment:
This code will fail because `k` is undefined until the for-loop below. 

Also in general it is better to give a meaningful name to a flag value, e.g here something like`is_numpyish` then the condition could read `if is_numpyish` which is more self-documenting than `if conditions[0]`.

But I think we will actually want to tackle this differently anyway. Instead of simply lumping all numpy and pandas cases together, what would be advised is to add a new branch inside the loop below:
```
    else if <<is pandas>>:
        # do pandas-specific operations
```
Apart from keeping the original operation intact, this will also be be necessary because pandas is *not* a require dependency of Bokeh. We cannot just assume that a user has it installed. The "else" condition above will also have to guard against that. We have a util for conditionally importing optional dependencies:
```
pd = import_optional('pandas')
```
Then the condition can be `else if pd and (...)` which will work whether pandas is installed or not.

Reply:
@bryevdv Thanks for your feedback! It was my mistake, the line `conditions=(...)` is supposed to be below `for k, v in  new_data.items():`. But anyway, as you and @rad83 said, it is better to approach it differently.

I imagine it would be something like:

`if <numpy>:
    (......)
else:
    pd = import_optional('pandas')
    if pd and isinstance(self[k], pd.core.series.Series) or isinstance(new_data[k], pd.core.series.Series):
        #pandas-specific operations`

Please correct me if I am wrong again!

Reply:
Pandas is already *optionally* imported on line 77.

Reply:
@rad83 Sorry, I didn't see that before!